units/patch_intent.go

```diff
@@ -1183,10 +1183,16 @@ func fetchTriggerProjectInfo(ctx context.Context, patchDoc *patch.Patch) (*model
 		if err != nil {
 			return nil, nil, nil, errors.Wrapf(err, "getting version at revision '%s'", patchDoc.Triggers.DownstreamRevision)
 		}
+		if v == nil {
+			return nil, nil, nil, errors.Errorf("version at revision '%s' not found", patchDoc.Triggers.DownstreamRevision)
+		}
 		project, pp, err = model.FindAndTranslateProjectForVersion(ctx, evergreen.GetEnvironment().Settings(), v, true)
-	} else {
-		v, project, pp, err = model.FindLatestVersionWithValidProject(patchDoc.Project, true)
+		if err != nil {
+			return nil, nil, nil, errors.Wrapf(err, "getting downstream version at revision '%s' to use for patch '%s'", patchDoc.Triggers.DownstreamRevision, patchDoc.Id.Hex())
+		}
+		return v, project, pp, nil
```

Comment:
Since the `if` block returns at the end, the `var` declarations at the top aren't needed anymore, and the `else` can be removed since everything past L1193 is the same as an `else` block anyways.